File: passwordvault/src/main/java/co/blustor/pwv/adapters/IconAdapter.java
package co.blustor.pwv.adapters;

import android.content.Context;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;

import co.blustor.pwv.R;
import co.blustor.pwv.utils.MyApplication;

public class IconAdapter extends BaseAdapter {
    private final Context mContext;

    public IconAdapter(Context context) {
        mContext = context;
    }

    @Override
    public int getCount() {
        return 69;
    }

    @Override
    public Object getItem(int position) {
        return position;
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Nullable
    @Override
    public View getView(int position, @Nullable View convertView, ViewGroup parent) {
        LayoutInflater layoutInflater = (LayoutInflater) mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        if (layoutInflater != null) {
            View view;
            if (convertView == null) {
                view = layoutInflater.inflate(R.layout.grid_icon, parent, false);
            } else {
                view = convertView;
            }

            ImageView iconImageView = view.findViewById(R.id.imageview_icon);
            iconImageView.setImageResource(MyApplication.getIcons().get(position));

            return view;
        }

        return null;
    }
}

File: gatekeeper/src/main/java/co/blustor/identity/adapters/IconAdapter.kt
package co.blustor.identity.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter

import co.blustor.identity.R
import co.blustor.identity.utils.MyApplication

import kotlinx.android.synthetic.main.grid_icon.view.*

class IconAdapter(private val mContext: Context) : BaseAdapter() {

    override fun getCount(): Int {
        return MyApplication.icons.size()
    }

    override fun getItem(position: Int): Any {
        return position
    }

    override fun getItemId(position: Int): Long {
        return 0
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View? {
        val layoutInflater = mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val view = convertView ?: layoutInflater.inflate(R.layout.grid_icon, parent, false)

        view.imageViewIcon.setImageResource(MyApplication.icons.get(position))

        return view
    }
}
